### ISSUE TITLE
Transport class does not use platform specific user and group

### ISSUE THREAD

**Issue by dzeleski** [sensu/sensu-puppet#838]
The following occurs on windows:
`change from 'NT AUTHORITY\SYSTEM' to 'sensu' failed: Could not find group sensu`

`/etc/puppetlabs/code/environments/lab/modules/sensu/manifests/transport.pp:21`

Looks like the transport class is not using the platform specific var?

https://github.com/sensu/sensu-puppet/blob/master/manifests/transport.pp#L23

Needs to look something like this:
```
  file { "${sensu::conf_dir}/transport.json":
    ensure  => $ensure,
    owner   => $::sensu::user,
    group   => $::sensu::group,
    mode    => '0440',
    content => inline_template('<%= JSON.pretty_generate(@transport_type_hash) %>'),
    notify  => $::sensu::check_notify,
}
```

**Comment by ghoneycutt** [comment]
Released in v2.40.0

**Comment by ghoneycutt** [comment]
Released in v2.40.0
